---
title: "Descriptive statistics"
subtitle: "Quantitative and graphical techniques for summarizing data"
format: 
  revealjs:
    logo: img/poly-logo-2.jpg
    footer: "STAT218"
    smaller: true
    mermaid:
      theme: neutral
execute: 
  echo: false
  warning: false
  message: false
---

## Today's agenda

1. \[lecture\] frequency distributions; measures of spread and center
2. \[lab\] descriptive statistics and simple graphics in R

## Last time

::: {.columns}

::: {.column width="60%"}

1. Data semantics

- **categorical** data: ordinal (ordered) or nominal (unordered)
- **numeric** data: continuous (no 'gaps') or discrete ('gaps')

2. Data types and data structures in R

- basic types: numeric, character, logical, integer
- a **vector** is a collection of values of one type
- a **data frame** is a type-heterogeneous list of vectors of equal length

:::

::: {.column width="40%"}

Vectors can store observations of one variable:
```{r, echo = T}
# 4 observations of age
ages <- c(18, 22, 18, 12)
ages
```

Data frames can store observations of many variables:
```{r, echo = T}
# 3 observations of 2 variables
data.frame(subject.id = c(11, 2, 31),
           age = c(24, 31, 17),
           sex = c('m', 'm', 'f'))
```

:::

:::

*Techniques for summarizing data depend on the data type*


## What are descriptive statistics?

We learned last time that a statistic is a data summary, *i.e.*, any function of a set of observations.

**Descriptive statistics** refers to analysis of sample characteristics using summary statistics.

- these are data analyses that uses statistics interpreted on face value
- in contrast to **inferential statistics**, which uses statistics interpreted relative to a broader population

Descriptive statistics can be either numerical or graphical; we'll discuss both.

## Dataset: FAMuSS study

Observational study of 595 individuals comparing change in arm strength before and after resistance training between genotypes for a region of interest on the ACTN3 gene.

> Pescatello, L. S., *et al.* (2013). Highlights from the functional single nucleotide polymorphisms associated with human muscle size and strength or FAMuSS study. *BioMed research international.*

```{r}
library(tidyverse)
library(oibiostat)
library(pander)
data(famuss)
head(famuss, 4) %>% pander(caption = 'Example data rows')
```

## Categorical frequency distributions

For categorical variables, the frequency distribution is simply an observation count by category. For example:

::: columns
::: {.column width="0.2"}
```{r}
set.seed(11424)
famuss %>% 
  mutate(participant.id = row_number()) %>%
  rename(genotype = actn3.r577x) %>%
  select(participant.id, genotype) %>% 
  sample_n(size = 6) %>% 
  pander(caption = 'Data table')
```
:::

::: {.column width="0.8"}
```{r}
tbl <- famuss %>% 
  mutate(participant = row_number()) %>%
  rename(genotype = actn3.r577x) %>%
  select(genotype) %>%
  table() 

tbl %>%
  pander(caption = 'Frequency distribution')
```

```{r}
par(mar = c(5, 5, 2, 2), cex.lab = 2, cex.axis = 1.5, cex = 1.5)
famuss %>% 
  mutate(participant = row_number()) %>%
  rename(genotype = actn3.r577x) %>%
  pull(genotype) %>%
  plot(xlab = 'genotype', ylab = 'frequency')
```
:::
:::

## Numeric frequency distributions

Frequency distributions of numeric variables are observation counts by *range*; a plot of a numeric frequency distribution is called a **histogram**.

::: columns
::: {.column width="0.3"}
```{r}
famuss %>%
  mutate(participant.id = row_number()) %>%
  select(participant.id, bmi) %>%
  sample_n(size = 6) %>%
  pander(caption = 'Data table')
```
:::

::: {.column width="0.7"}
```{r}
par(mar = c(4, 5, 2, 2), 
    cex = 2.5)
famuss %>%
  transmute(bmi.range = cut(bmi, breaks = seq(10, 50, by = 10))) %>%
  table() %>%
  pander(caption = 'Frequency distribution')
hist(famuss$bmi, 
     breaks = 3, 
     main = 'Histogram', 
     xlab = 'bmi',
     ylab = 'frequency')
```
:::
:::

The operation of dividing a numeric variable into interval ranges is called **binning**.

## Histograms and binning

Binning has a big effect on the visual impression. Which one captures the shape best?

```{r}
layout(matrix(1:4, nrow = 2, byrow = T))
par(mar = c(5, 5, 2, 2), cex.lab = 2, cex.axis = 1.5, cex.main = 2)
hist(famuss$bmi, 
     breaks = 4,
     xlab = 'bmi',
     ylab = 'frequency',
     main = '4 bins',
     xlim = c(10, 50),
     ylim = c(0, 450))
hist(famuss$bmi, 
     breaks = 6,
     ylim = c(0, 450),
     xlim = c(10, 50),
     xlab = 'bmi',
     ylab = 'frequency',
     main = '6 bins')
hist(famuss$bmi, 
     xlim = c(10, 50),
     ylim = c(0, 450),
     breaks = 15,
     xlab = 'bmi',
     ylab = 'frequency',
     main = '15 bins')
hist(famuss$bmi, 
     xlim = c(10, 50),
     ylim = c(0, 450),
     breaks = 58,
     xlab = 'bmi',
     ylab = 'frequency',
     main = '58 bins')
```

## Shapes

For numeric variables, the histogram reveals the **shape** of the distribution:

-   **symmetric** if it shows left-right symmetry about a central value
-   **skewed** if it stretches farther in one direction from a central value

```{r, fig.height=3}
par(mfrow = c(1, 3))
curve(dgamma(abs(10 - x), shape = 3, scale = 1), 
      from = 0, 
      to = 10,
      xlab = '', 
      ylab = '', 
      main = 'left skew', 
      cex.main = 3, 
      xaxt = 'n', 
      yaxt = 'n')
curve(dnorm, 
      from = -3, 
      to = 3,
      xlab = '', 
      ylab = '', 
      main = 'symmetric', 
      cex.main = 3,
      xaxt = 'n', 
      yaxt = 'n')
curve(dgamma(x, shape = 3, scale = 1), 
      from = 0, 
      to = 10,
      xlab = '', 
      ylab = '', 
      main = 'right skew', 
      cex.main = 3, 
      xaxt = 'n', 
      yaxt = 'n')
```

## Modes

Histograms also reveal the number of **modes** or local peaks of frequency distributions.

-   **uniform** if there are zero peaks
-   **unimodal** if there is one peak
-   **bimodal** if there are two peaks
-   **multimodal** if there are two or more peaks

```{r, fig.height=3}
par(mfrow = c(1, 4))

curve(dunif, 
      from = -1, 
      to = 2,
      xlab = '', 
      ylab = '', 
      main = 'uniform', 
      cex.main = 3, 
      xaxt = 'n', 
      yaxt = 'n')
curve(0.4*dnorm(x, mean = 0) + 0.6*dnorm(x, mean = 2), 
      from = -3, 
      to = 6,
      xlab = '', 
      ylab = '', 
      main = 'unimodal', 
      cex.main = 3,
      xaxt = 'n', 
      yaxt = 'n')
curve(0.4*dnorm(x, mean = -2) + 0.6*dnorm(x, mean = 2), 
      from = -5, 
      to = 5,
      xlab = '', 
      ylab = '', 
      main = 'bimodal', 
      cex.main = 3, 
      xaxt = 'n', 
      yaxt = 'n')
curve(0.2*dnorm(x, mean = -2) + 
        0.5*dnorm(x, mean = 2) + 
        0.3*dnorm(x, mean = 6), 
      from = -5, 
      to = 10,
      xlab = '', 
      ylab = '', 
      main = 'multimodal', 
      cex.main = 3, 
      xaxt = 'n', 
      yaxt = 'n')
```

## Your turn: characterizing distributions

Consider four variables from the FAMuSS study. Describe the shape and modality.

```{r}
par(mfrow = c(2, 2),
    mar = c(5, 5, 2, 2), 
    cex.lab = 2, 
    cex.axis = 1.5, 
    cex.main = 2)
hist(famuss$height, 
     main = '', 
     xlab = 'height',
     ylab = 'frequency')
hist(famuss$weight, 
     main = '', 
     xlab = 'weight',
     ylab = 'frequency')
hist(famuss$age, 
     main = '', 
     xlab = 'age',
     ylab = 'frequency')
hist(famuss$bmi, 
     main = '', 
     xlab = 'bmi',
     ylab = 'frequency')
```

## Your turn: characterizing distributions

Here are some made-up data. Describe the shape and modality.

```{r}
par(mfrow = c(2, 2))
par(mar = c(5, 5, 2, 2), cex.lab = 2, cex.axis = 1.5, cex.main = 2)
set.seed(11524)
runif(1000, min = -5, max = 2) %>% 
  hist(main = '', xlab = 'x', ylab = 'frequency')
-rpois(1000, lambda = 2) %>% 
  hist(main = '', xlab = 'x', ylab = 'frequency', breaks = 6)
c(rnorm(400, mean = -4, sd = 2), rnorm(600, mean = 8, sd = 5)) %>%
  hist(main = '', xlab = 'x', ylab = 'frequency', breaks = 20)
c(rexp(500, rate = 1), 10 - rexp(600, rate = 2))  %>%
  hist(main = '', xlab = 'x', ylab = 'frequency', breaks = 20)
```


## Descriptive measures

A **descriptive measure** is a summary statistic that captures a particular feature of the frequency distribution of a numeric variable.

Commonly, measures capture either **location** or **spread**.

::: {.columns}

::: {.column}
Measures of location:

- mean
- median
- mode
- percentiles/quantiles
:::

::: {.column}
Measures of spread:

- range (min and max)
- interquartile range
- average deviation
- variance
- standard deviation
:::

:::

It is common practice to report multiple measures.

## Measures of location

Often location is specified by the "center" of a frequency distribution.

::: columns
::: {.column width="0.5"}
There are three common measures of center, each of which corresponds to a slightly different meaning of "typical":

| Measure | Definition          |
|---------|---------------------|
| Mode    | Most frequent value |
| Mean    | Average value       |
| Median  | Middle value        |
:::

::: {.column width="0.5"}
Suppose your data consisted of the following observations of age in years:

```{r}
c(19, 19, 21, 25, 31) %>% pander()
```

-   the **mode** or most frequent value is 19
-   the **median** or middle value is 21
-   the **mean** or average value is $\frac{19 + 19 + 21 + 25 + 31}{5}$ = 23
:::
:::

## Quick example

Consider the first 8 observations of change in nondominant arm strength from the FAMuSS study data:

```{r}
famuss$ndrm.ch[1:8] %>% pander()
```

Compute the mean, median, and mode.

## Comparing measures of center

Each statistic is a little different, but often they roughly agree; for example, all are between 20 and 25, which seems to capture the typical BMI well enough.

```{r, fig.width = 10, fig.height = 4}
makehist <- function(x, b, t, xlab, leg, nmodes){
h <- hist(x, 
          main = b, 
          breaks = t, 
          xlab = xlab,
          ylab = 'frequency')
abline(v = mean(x), 
       col = 2, 
       lty = 2, 
       lwd = 2)
abline(v = median(x),
       col = 4, 
       lty = 3, 
       lwd = 2)
x_mode <- h$mids[which(h$counts %in% 
                         sort(h$counts, 
                              decreasing = T)[1:nmodes])]
abline(v = x_mode, 
       col = 6, 
       lty = 4, 
       lwd = 2)
if(leg){
legend(quantile(h$mids, 0.7), 
       max(h$counts), 
       legend = c("Mean", "Median", "Mode"), 
       col = c(2, 4, 6), 
       lty = c(2, 3, 4))
}
}
par(mar = c(4, 5, 2, 2), 
    cex.lab = 2, 
    cex.axis = 1.5,
    cex.legend = 1.5,
    cex.main = 2)
makehist(famuss$bmi, '', 15, 'bmi', T, 1)
```

*How do you think the frequency distribution affects which one is "best"?*

## Means, medians, and skewness

The mean and median both get 'pulled' in the direction of skewness, but the mean is more sensitive:

```{r, fig.width = 20, fig.height = 4}
set.seed(11524)
x1 <- rgamma(1000, shape = 3, scale = 2)
x2 <- -rgamma(1000, shape = 3, scale = 2)

par(mfrow = c(1, 2),
    cex = 2,
    mar = c(2, 4, 1, 1))
makehist(x1, '', 20, '', T, 1)
makehist(x2, '', 20, '', F, 1)
```

Comparing means and medians captures information about skewness present since:

- mean $>$ median: right skew
- mean $<$ median: left skew
- mean $\approx$ median: symmetric

## When to use mode(s)

Mode is rarely used unless extreme skewness or multiple modes are present; below are two examples.

```{r, fig.width = 20, fig.height = 6}
set.seed(11524)

x_skew <- c(rgamma(n = 1000, shape = 4, rate = 1/10),

            runif(n = 500, min = 70, max = 300))

x_bi <- c(rnorm(1000, mean = -3),

          rnorm(1000, mean = 3))

par(mfrow = c(1, 2),
    cex = 2)

makehist(x_skew, '', 20, '', T, 1)

makehist(x_bi, '', 20, '', F, 2)
```

## Percentiles

A **percentile** is a value with specified proportions of data lying both above and below that value. 

- measure of location (but not center)
- defined with reference to the percentage of data below

For example, the 20th percentile is the value with 20% of observations below and 80% of observations above. Suppose we have 5 observations:

```{r}
x <- c(19, 20, 21, 25, 31)
rbind(age = x, rank=rank(x)) %>% pander()
```

The 20th percentile is not unique! In fact *any* number between 19 and 20 is a 20th percentile since it would satisfy:

-   20% below (19)
-   80% above (20, 21, 25, 31)


## Cumulative frequency distribution

The *cumulative frequency distribution* is a data summary showing percentiles. Think of it as percentile (y) against value (x).

::: columns
::: {.column width="0.6"}
```{r, fig.height = 5, fig.width = 5, fig.align='center'}
par(mar = c(5, 5, 2, 2), cex.lab = 2, cex.axis = 1.5, cex.main = 2)
famuss$age[famuss$age < 30] %>%
  # sample(size = 50) %>%
  ecdf() %>% 
  plot(main = '', 
       xlab = 'age', 
       ylab = 'cumulative frequency')
```
:::

::: {.column width="0.4"}
Interpretation of some specific values:

-   about 40% of the subjects are 20 or younger
-   about 80% of the subjects are 24 or younger

*Your turn*:

1.  Roughly what percentage of subjects are 22 or younger?
2.  About what age is the 10th percentile?
:::
:::


## Common percentiles

::: columns
::: {.column width="50%"}
The **five-number summary** is a collection of five percentiles that succinctly describe the frequency distribution:

| Statistic name     | Meaning          |
|--------------------|------------------|
| **minimum**        | 0th percentile   |
| **first quartile** | 25th percentile  |
| **median**         | 50th percentile  |
| **third quartile** | 75th percentile  |
| **maximum**        | 100th percentile |
:::

::: {.column width="50%"}
Boxplots provide a graphical display of the five-number summary.

![](img/boxplot-anatomy.png)
:::

:::

## Boxplots vs. histograms

Notice how the two displays align, and also how they differ. The histogram shows shape in greater detail, but the boxplot is much more compact.

::: columns
::: {.column width="50%"}
```{r, fig.width=5, fig.height=5}
par(mar = c(4.5, 5, 2, 2), cex.lab = 2, cex.axis = 1.5, cex.main = 2)
layout(matrix(1:2, nrow = 2), height = c(3, 2))
hist(famuss$ndrm.ch, 
     main = 'nondominant arm', 
     xlab = 'percent change',
     ylab = 'frequency')
boxplot(famuss$ndrm.ch, 
        range = 3, 
        horizontal = T)
```

:::

::: {.column width="50%"}
```{r, fig.width=5, fig.height=5}
par(mar = c(4.5, 5, 2, 2), cex.lab = 2, cex.axis = 1.5, cex.main = 2)
layout(matrix(1:2, nrow = 2), height = c(3, 2))
hist(famuss$drm.ch, 
     main = 'dominant arm', 
     xlab = 'percent change',
     ylab = 'frequency')
boxplot(famuss$drm.ch, 
        range = 3, 
        horizontal = T,
        ylim = c(-40, 100))
```
:::
:::


## Measures of spread

The *spread* of observations refers to how concentrated or diffuse the values are.

```{r, fig.width = 8, fig.height = 2.5, fig.align='center'}
par(mfrow = c(1, 2))
par(mar = c(1, 1, 4, 1),
    cex.main = 2)
curve(dnorm, from = -3, to = 3, 
      main = 'more spread', 
      xaxt = 'n', 
      yaxt = 'n',
      xlab = '',
      ylab = '')
curve(dnorm(x, sd = 0.25), from = -3, to = 3, 
      main = 'less spread', 
      xaxt = 'n', 
      yaxt = 'n',
      xlab = '',
      ylab = '')
```

Two ways to understand and measure spread:

-   *ranges* of values capturing much of the distribution
-   *deviations* of values from a central value

## Range-based measures

A simple way to understand and measure spread is based on ranges. Consider more ages, sorted and ranked:

```{r}
x <- c(16, 18, 19, 20, 21, 22, 25, 26, 28, 29, 30, 34)
rbind(age = x, rank = rank(x)) %>% pander()
```

-   The **range** is the minimum and maximum values: $$\text{range} = (\text{min}, \text{max}) = (16, 34)$$

-   The **interquartile range** (IQR) is the difference \[75th percentile\] - \[25th percentile\] $$\text{IQR} = 29 - 19 = 10$$ *When might you prefer IQR to range? Can you think of an example?*

## Deviation-based measures

Another way is based on *deviations* from a central value. Continuing the example, the mean age is is 24. The deviations of each observation from the mean are:

```{r}
ages <- c(16, 18, 19, 20, 21, 22, 25, 26, 28, 29, 30, 34)

rbind(age = ages, deviation = ages - mean(ages)) %>% pander()
```

The **average deviation** is defined as the average of the absolute values of the deviations from the mean: $$\frac{8 + 6 + 5 + 4 + 3 + 2 + 1 + 2 + 4 + 5 + 6 + 10}{12}$$

## Deviation-based measures

Another way is based on *deviations* from a central value. Continuing the example, the mean age is is 24. The deviations of each observation from the mean are:

```{r}
ages <- c(16, 18, 19, 20, 21, 22, 25, 26, 28, 29, 30, 34)

rbind(age = ages, deviation = ages - mean(ages)) %>% pander()
```

The **variance** is the average *squared* deviation from the mean (but divided by one less than the sample size): $$\frac{(-8)^2 + (-6)^2 + (-5)^2 + (-4)^2 + (-3)^2 + (-2)^2 + (1)^2 + (2)^2 + (4)^2 + (5)^2 + (6)^2 + (10)^2}{12 - 1}$$

The **standard deviation** is the square root of the variance: $$\sqrt{\frac{(-8)^2 + (-6)^2 + (-5)^2 + (-4)^2 + (-3)^2 + (-2)^2 + (1)^2 + (2)^2 + (4)^2 + (5)^2 + (6)^2 + (10)^2}{12 - 1}}$$

## Mathematical notations

Following the convention from before, write a set of $n$ observations as $x_1, x_2, \dots, x_n$.

::: {.columns}

::: {.column}

The **mean** of the observations is written: $$\bar{x} = \frac{1}{n}\sum_i x_i$$

The **average deviation** is: $$\frac{1}{n} \sum_i |x_i - \bar{x}|$$

:::

::: {.column}

The **variance** is: $$s_x^2 = \frac{1}{n - 1}\sum_i (x_i - \bar{x})^2$$

The **standard deviation** is: $$s_x = \sqrt{\frac{1}{n - 1}\sum_i (x_i - \bar{x})^2}$$

:::

:::
## Interpretations

Listed from largest to smallest, here are each of the measures of spread for the 12 ages:

```{r}
c(min = min(ages),
  max = max(ages),
  iqr = IQR(ages),
  variance = var(ages),
  st.dev = sd(ages),
  avg.dev = mean(abs(ages - mean(ages)))) %>%
  pander()
```

The interpretations differ between these statistics:

-   \[range\] all of the data lies on an between 16 and 34 years old on an interval 18 years in width
-   \[IQR\] the middle half of the data lies on an interval 8.5 years in width
-   \[average deviation\] the average distance from the mean is 4.67 years
- \[variance\] the average squared distance from the mean is 30.55 years$^2$
-   \[standard deviation\] the average squared distance from the mean, rescaled to years, is 5.53 years


